utils.js

```diff
@@ -20,12 +30,17 @@ CTLEventUtils.formatShortDate = function(d) {
  * @return = an array of events that match the query
  */
 CTLEventUtils.searchEvents = function(allEvents, index, q) {
-    var results = index.search(q);
+    if (q == null || q == '') {
```

Comment:
`if (q)` should serve the same purpose here. See [Logical Operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators)

also - question on Ln. 70, `if (audience === null || audience === 'null') `. Could audience evaluate to 'null'?

Reply:
I don't think it could, but I've kept it around to standardize the filter behavior - if there are no matches it returns everything. As I'm writing this out though, this seems like an unexpected behavior.  Should I remove them?

Reply:
A rule of thumb I follow is to always use triple-equal in javascript boolean statements. `(q == null || q == '')` is redundant because, since the empty string is `==` to null, the second clause here doesn't do anything.

If you are just trying to find out if the value is "falsey", `if (q)` should suffice, like Susan points out. Otherwise, explicitly using `===` will make your code clearer.

Reply:
on the audience question - I think the shortcut of `if (null) { return all }` is fine. I was more curious about the test for a string with `'null'`. Is that a possible value?

Reply:
Oh, that's actually incorrect. In my browser, `'' == null` is actually false. Just when you think you understand it, javascript can do something unexpected.

But anyways, always using `===` is still a good rule of thumb.

Reply:
@sdreher No, its not, but it was prior to refactoring. I was doing something silly like this: https://github.com/ccnmtl/calendar-widget/pull/80/files#diff-a8a494a0e94448e7e4cce52d0cb1ce9eL42